usr/share/rear/restore/DUPLICITY/default/400_restore_duplicity.sh

```diff
@@ -31,7 +31,7 @@ if [ "$BACKUP_PROG" = "duplicity" ]; then
 
     # ensure we have enougth space to unpack the backups (they are 100M, but neet up to 1G to unpack!)
     mkdir -p /mnt/tmp
-    mount -t tmpfs none /mnt/tmp
+    mount -t tmpfs none /mnt/tmp -o size=8G
```

Comment:
@Lukey3332 `size=8G` is maybe good for your case, but for others it might be better to work with percentage.
According the man page the size is given in bytes, and rounded up to entire pages.  The default  is  half  of  the memory. The size parameter also accepts a suffix % to limit this tmpfs instance to that  percentage of your physical RAM: the default, when neither size nor nr_blocks is specified, is size=50%

Reply:
Yes, I already tried with `size=100%`, but on an 4Gb Ram Machine this was not enough for me. So with this (+ Manually mounting Swap during restore) it worked better. However more testing showed that in Worst Case, this needs to be as big as the biggest File in the Backup (So for me it was `size=40G` in the and + Enough Swap on an external HDD). Any Ideas on how to Solve this Problem? Maybe creating an Temporary Directory on the biggest Target disk?

Reply:
@Lukey3332 
in general avoid hardcoded things like
<pre>
mount ... -o size=8G
</pre>
when that does not work on really any system.

System-specific settings must have a documented
variable in default.conf so that the user can see
that such a setting exists and that he may have
to adapt that as he needs in his particular case,
for example have a look at
https://github.com/rear/rear/issues/1213#issuecomment-285618629